Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp

```diff
@@ -709,12 +790,36 @@ Protocol::ErrorStringOr<void> InspectorDebuggerAgent::addSymbolicBreakpoint(cons
         JSC::JSLockHolder locker(m_debugger.vm());
 
         m_debugger.forEachRegisteredCodeBlock([&] (JSC::CodeBlock* codeBlock) {
-            if (symbolicBreakpoint.matches(functionName(codeBlock)))
+            if (symbolicBreakpoint.matches(functionName(*codeBlock)))
                 codeBlock->addBreakpoint(1);
         });
     }
 
-    // FIXME: <https://webkit.org/b/243716> Web Inspector: Debugger: symbolic breakpoints should work with native functions
+#if ENABLE(JIT)
+    Vector<JSC::NativeExecutable*> newNativeExecutables;
+    {
+        Locker locker { s_replacedThunksLock };
+        auto& existingReplacedThunks = replacedThunks();
+
+        JSC::HeapIterationScope iterationScope(m_debugger.vm().heap);
+        m_debugger.vm().heap.objectSpace().forEachLiveCell(iterationScope, [&] (JSC::HeapCell* cell, JSC::HeapCell::Kind kind) {
+            if (isJSCellKind(kind)) {
+                if (auto* nativeExecutable = JSC::jsDynamicCast<JSC::NativeExecutable*>(static_cast<JSC::JSCell*>(cell))) {
+                    if (auto existingIndex = existingReplacedThunks.find(nativeExecutable); existingIndex != notFound)
+                        ++existingReplacedThunks[existingIndex]->matchCount;
+                    else
+                        newNativeExecutables.append(nativeExecutable);
+                }
+            }
+
+            return IterationStatus::Continue;
+        });
+    }
+    for (auto* nativeExecutable : WTFMove(newNativeExecutables))
+        didCreateNativeExecutable(*nativeExecutable);
```

Comment:
Let's do this inside `DeferGCForAWhile deferGC(m_vm)` scope. `newNativeExecutables` is not heap reachable vector, so if GC happens inside `didCreateNativeExecutable`, we will do GC crash.